x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/CompositeAggsRowSet.java

```diff
@@ -40,18 +40,22 @@
         }
 
         // page size
-        size = limit < 0 ? buckets.size() : Math.min(buckets.size(), limit);
+        size = limit == -1 ? buckets.size() : Math.min(buckets.size(), limit);
```

Comment:
It will also work, but we introduced the `-1` which means "no limit - need to process all buckets", and it's the only negative value we can get now. If you check [here](https://github.com/elastic/elasticsearch/pull/43154/files/8ea19efa6df6ca80b02d31d61f05b598d7b4d854#diff-c4c82fd30c72e84bb90b408910d43a4eR52) you'll see that when we want to stop we set remaining limit to `0`. So to sum up, I changed the check to `== -1` for more clarity.